nautobot/dcim/factory.py

```diff
@@ -0,0 +1,177 @@
+
+    has_description = factory.Faker("pybool")
+    description = factory.Maybe("has_description", factory.Faker("text", max_nb_chars=200), "")
+
+    @factory.lazy_attribute
+    def parent(self):
+        if self.name in ["Floor", "Elevator"]:
+            return LocationType.objects.get(name="Building")
+        if self.name == "Room":
+            return LocationType.objects.get(name="Floor")
+        if self.name == "Aisle":
+            return LocationType.objects.get(name="Room")
+        return None
+
+    @factory.post_generation
+    def content_types(self, create, extract, **kwargs):
+        """Assign some contenttypes to a location after generation"""
+        if self.name in ["Root"]:
+            self.content_types.set(ContentType.objects.filter(FeatureQuery("locations").get_query()))
+        elif self.name in ["Building", "Floor"]:
+            self.content_types.set(
+                [
+                    ContentType.objects.get_for_model(Prefix),
+                    ContentType.objects.get_for_model(Rack),
+                    ContentType.objects.get_for_model(RackGroup),
+                    ContentType.objects.get_for_model(VLANGroup),
+                    ContentType.objects.get_for_model(VLAN),
+                ]
+            )
+        elif self.name in ["Room", "Elevator"]:
+            self.content_types.set(
+                [
+                    ContentType.objects.get_for_model(Cluster),
+                    ContentType.objects.get_for_model(PowerPanel),
+                    ContentType.objects.get_for_model(VLAN),
+                ]
+            )
+        elif self.name in ["Aisle"]:
+            self.content_types.set(
+                [
+                    ContentType.objects.get_for_model(CircuitTermination),
+                    ContentType.objects.get_for_model(Device),
+                    ContentType.objects.get_for_model(PowerPanel),
+                    ContentType.objects.get_for_model(VLAN),
+                ]
+            )
+
+
+class LocationFactory(DjangoModelFactory):
+    class Meta:
+        model = Location
+        exclude = (
+            "has_parent",
+            "has_site",
+            "has_tenant",
+            "has_description",
+        )
+
+    name = factory.LazyAttributeSequence(lambda l, n: f"{l.location_type.name}-{n:02d}")
+    status = random_instance(lambda: Status.objects.get_for_model(Location), allow_null=False)
+
+    has_parent = factory.LazyAttribute(lambda l: bool(l.location_type.parent))
+    parent = factory.Maybe(
+        "has_parent",
+        factory.LazyAttribute(
+            lambda l: factory.random.randgen.choice(Location.objects.filter(location_type=l.location_type.parent))
+            if Location.objects.count()
+            else None
+        ),
+    )
+
+    has_site = factory.LazyAttribute(lambda l: not bool(l.has_parent))
+    site = factory.Maybe("has_site", random_instance(Site))
```

Comment:
Should all `random_instance()` calls have `allow_null` set to `False`?

Reply:
Only where it's a required field that cannot be null. Otherwise I think it's fine to quietly fall through to null in the case where no related objects exist.